src/Container.php

```diff
@@ -319,6 +319,11 @@ public function setInstance($name, $instance) {
     public function addCall($method, array $args = []) {
         $this->currentRule['calls'][] = [$method, $args];
 
+        // Something added a rule. If we have any existing factories make sure we clear them.
+        if (isset($this->factories[$this->currentRuleName])) {
+            unset($this->factories[$this->currentRuleName]);
```

Comment:
I think you can actually omit the if statement. You are allowed to unset something that doesn't exist without fail.